functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+		[switch]$Scripts,
+        [Parameter(ParameterSetName="Paths")]
+        [Switch]$XpDirTree,
+        [Parameter(ParameterSetName="Paths")][Parameter(ParameterSetName="Files")]
+        [switch]$ScriptOnly,
+        [Parameter(ParameterSetName="Paths")][Parameter(ParameterSetName="Files")]
+		[switch]$VerifyOnly,
+        [Parameter(ParameterSetName="Paths")][Parameter(ParameterSetName="Files")]
+        [switch]$OlaStyle,
+        [Parameter(ParameterSetName="Paths")][Parameter(ParameterSetName="Files")]
+		[string]$LogicalFilePrefix,
+        [Parameter(ParameterSetName="Paths")][Parameter(ParameterSetName="Files")]
+		[hashtable]$LogicalFileMapping			
+	)
+    BEGIN
+    {
+        $FunctionName = "Restore-DbaBackup"
+        $BackupFiles = @()
+    }
+    PROCESS
+    {
+        if ($PSCmdlet.ParameterSetName -eq "Paths")
+        {
+            Write-Verbose "$FunctionName : Paths passed in" 
+            foreach ($p in $path)
+            {  
+                if ($XpDirTree)
+                {
+                    $BackupFiles += Get-XPDirTreeRestoreFile -path $p -SqlServer $SqlServer -SqlCredential $SqlCredential
+                }
+                elseif ((Get-Item $p).PSIsContainer -ne $true)
+                {
+                    Write-Verbose "$FunctionName : Single file"
+                    $BackupFiles += Get-item $p
+                } 
+                elseif ($OlaStyle)
+                {
+                    Write-Verbose "$FunctionName : Ola Style"
+                    $BackupFiles += Get-OlaHRestoreFile -path $p
+                } 
+                else 
+                {
+                    Write-Verbose "$FunctionName : Standard Directory"
+                    $BackupFiles += Get-DirectoryRestoreFile -path $p
+                }
+            }
+        }elseif($PSCmdlet.ParameterSetName -eq "Files")
+        {
+            Write-Verbose "$FunctionName : Files passed in $($Files.count)" 
+            Foreach ($File in $Files)
+            {
+                $BackupFiles += $File
+            }
+        }
+    }
+    END
+    {
+        $FilteredFiles = $BackupFiles | Get-FilteredRestoreFile -SqlServer $SqlServer -RestoreTime $RestoreTime -SqlCredental $SqlCredential
+
+        if (($FilteredFiles.DatabaseName | Group-Object | Measure-Object).count -gt 1)
+        {
+            $dbs = ($FilteredFiles | Select DatabaseName) -join (',')
+            Write-Error "$FunctionName - We can only handle 1 Database at a time - $dbs"
+        }
+        if(Test-DbaLsnChain -FilteredRestoreFiles $FilteredFiles)
+        {
+            try{
+                $FilteredFiles | Restore-DBFromFilteredArray -SqlServer $SqlServer -DBName $databasename -SqlCredential $SqlCredential -RestoreTime $RestoreTime -RestoreLocation $RestoreLocation -NoRecovery:$NoRecovery -ReplaceDatabase:$ReplaceDatabase -Scripts:$Scripts -ScriptOnly:$ScriptOnly -VerifyOnly:$VerifyOnly
```

Comment:
Restore-DBFromFilteredArray accepts -FileStructure or -RestoreLocation, but when called here, we can only pass $RestoreLocation? That means all data and log files will be in one directory?